src/mozillavpn.cpp

```diff
@@ -1432,9 +1430,6 @@ void MozillaVPN::subscriptionFailedInternal(bool canceledByUser) {
         ErrorHandler::SubscriptionFailureError);
   }
 
-  TaskScheduler::scheduleTask(
```

Comment:
@bakulf need your help with this deletion.

The problem that I was finding was that when I intentionally cancelled the web subscription this new TaskAccount causes accountChecked to run which then causes the emptying of the settings holder. This caused subsequent restarts of my subscribe task to fail.

This change also seems to resolve VPN-3217 (canceling and then resubscribing causes error in subscription screen).

So, removing it seems like a win in these two cases. However, calling TaskAccount in subscriptionFailedInternal was added by you a *long* time ago (https://github.com/mozilla-mobile/mozilla-vpn-client/commit/234345d89ad362024254a6d0061bc1ca208847d7). Removing code that's been around for so long makes me nervous. Would love a consult from you on workflows where we would want this to happen and your thoughts on this change.

Reply:
I see why this fixes a lot of issues, and I'm happy with this change. The only concern I have is that, if the user completes the subscription on another device, there is no way to resume from this view except closing the app and restart. But this can be fixed in a follow up.